symtab_builder.cc

```diff
@@ -49,7 +49,7 @@ bool SymtabBuilder::Resolve(const std::string& name, const std::string& soname,
         Syminfo* found = NULL;
         for (int i = 0; i < src_syms_.size(); i++) {
             if (src_syms_[i].name == name && src_syms_[i].soname == soname && src_syms_[i].version == version) {
-                found = &src_syms_[i];
+                if (found == NULL || IsDefined(*src_syms_[i].sym)) found = &src_syms_[i];
```

Comment:
Hmm I'm not sure if this is right. When there are multiple defined symbols, shouldn't the first symbol win?

I think I'd suggest we should change back `src_syms_` from vector to map (keyed by a tuple of name, soname, and version)

Reply:
> shouldn't the first symbol win?
I think so.
> change back src_syms_ from vector to map
That sounds good.